### ISSUE TITLE
Discover BLE on background

### ISSUE THREAD

**Issue by ninamanalo19** [AXAET/iBeaconSDK2-example-iOS#2]
can you give a sample snippet for this one? Thanks
not for ranging but discovering AXABeacon itself

**Comment by AXAET** [comment]
first, you must check the TARGETS -> Capabilities -> Background Modes -> and select Uses Bluetooth LE accessories and select Acts as a Bluetooth LE accessories.
then, you can call startFindBleDevicesWithServices: options: method and specify an array of services UUIDs to discover peripheral on background.
for example, you can open the iBeaconSDK2-example-iOS and select TagViewController.m , and in the viewWillAppear method, you try to delete "[[AXABeaconManager sharedManager] startFindBleDevices]" and open the line at the end of viewWillAppear. then , replace the @"FFF0" to your Ble service UUID.

**Comment by ninamanalo19** [comment]
![img_3011](https://cloud.githubusercontent.com/assets/16302770/19549054/0b8a8f1c-96b3-11e6-92da-f9ef127749bc.JPG)
Hi Thanks for the response. To be sure, are the red ones the UUID that I can replace with FFF0?

**Comment by AXAET** [comment]
NO, the red ones UUID is proximityUUID of CLBeacon, not the serviceUUID, the @"FFF0" is a uuid of service in the Broadcast packets of a Ble,  but if the Broadcast packet haven't  a service UUID,  you couldn't discover a Ble on background.  it's better for you to look the  document of apple https://developer.apple.com/library/content/documentation/NetworkingInternetWeb/Conceptual/CoreBluetooth_concepts/CoreBluetoothOverview/CoreBluetoothOverview.html#//apple_ref/doc/uid/TP40013257-CH2-SW1.

**Comment by ninamanalo19** [comment]
Thank you.

**Comment by ninamanalo19** [comment]
I tried this `[[AXABeaconManager sharedManager] startFindBleDevicesWithServices:nil options:@{CBCentralManagerScanOptionAllowDuplicatesKey : @YES}]` under `viewWillAppear`
(I tried `nil` for testing)
but, when the app comes in background, it still doesn't work in background

here's my background modes
<img width="664" alt="screen shot 2016-10-20 at 12 15 08 pm" src="https://cloud.githubusercontent.com/assets/16302770/19552018/0f9ff1bc-96bf-11e6-8120-8b474f8962ad.png">

**Comment by ninamanalo19** [comment]
Nevermind, looks like from my android side it shows this log

`10-20 12:46:04.386: D/BluetoothLeScanner(13265): onScanResult() - ScanResult{mDevice=EC:11:27:29:B1:E0, mScanRecord=ScanRecord [mAdvertiseFlags=6, mServiceUuids=null, mManufacturerSpecificData={76=[2, 21, -3, -91, 6, -109, -92, -30, 79, -79, -81, -49, -58, -21, 7, 100, 120, 37, 39, 20, -44, -48, -64]}, mServiceData={}, mTxPowerLevel=-2147483648, mDeviceName=carToyota��], mRssi=-60, mTimestampNanos=262423995818132}`

which shows `mServiceUuids=null`

**Comment by AXAET** [comment]
I think it couldn't be scanned in the background. Because if there is no serviceUUID in the broadcast packet, it can not be scanned in the background as indicated in Apple's official document about CoreBluetooth. it's said you must specify its serviceUUID when scan in the background, so if this is the case, it will not be scanned in the background.

**Comment by ninamanalo19** [comment]
understood. thank you for clarifying my questions. I will follow up once I have new questions. have a good day!
